src/bun.js/bindings/BunString.cpp

```diff
@@ -467,6 +467,17 @@ extern "C" BunString URL__getFileURLString(BunString* filePath)
     return Bun::toStringRef(WTF::URL::fileURLWithFileSystemPath(filePath->toWTFString()).stringWithoutFragmentIdentifier());
 }
 
+extern "C" size_t URL__originLength(BunString* str)
+{
+    WTF::String string = str->toWTFString();
+    if (!string)
+        return 0;
+    WTF::URL url(string);
+    if (!url.isValid())
+        return 0;
+    return url.pathStart();
```

Comment:
If the URL contains non-ascii characters or gets puny decoded, this will return a UTF-16 byte length instead of a UTF-8 byte length

rihgt?

Reply:
true

Reply:
the new version of this function only operates on ascii